src/datastore/pg-store-v2.ts

```diff
@@ -629,12 +629,52 @@ export class PgStoreV2 extends BasePgStoreModule {
       if (cycleCheck.count === 0)
         throw new InvalidRequestError(`PoX cycle not found`, InvalidRequestErrorType.invalid_param);
       const results = await sql<(DbPoxCycleSigner & { total: number })[]>`
-        SELECT
-          signing_key, weight, stacked_amount, weight_percent, stacked_amount_percent,
-          COUNT(*) OVER()::int AS total
-        FROM pox_sets
-        WHERE canonical = TRUE AND cycle_number = ${args.cycle_number}
-        ORDER BY weight DESC, stacked_amount DESC, signing_key
+        WITH signer_keys AS (
+            SELECT DISTINCT ON (stacker) stacker, signer_key
+            FROM pox4_events
+            WHERE canonical = true AND microblock_canonical = true
+                AND name in ('stack-aggregation-commit-indexed', 'stack-aggregation-commit')
+                AND start_cycle_id <= ${args.cycle_number}
+                AND end_cycle_id >= ${args.cycle_number}
+            ORDER BY stacker, block_height DESC, tx_index DESC, event_index DESC
+        ), delegated_stackers AS (
+            SELECT DISTINCT ON (main.stacker) 
+                main.stacker, 
+                sk.signer_key
+            FROM pox4_events main
+            LEFT JOIN signer_keys sk ON main.delegator = sk.stacker
+            WHERE main.canonical = true 
+                AND main.microblock_canonical = true
+                AND main.name IN ('delegate-stack-stx', 'delegate-stack-increase', 'delegate-stack-extend')
+                AND main.start_cycle_id <= ${args.cycle_number} 
+                AND main.end_cycle_id >= ${args.cycle_number}
+            ORDER BY main.stacker, main.block_height DESC, main.microblock_sequence DESC, main.tx_index DESC, main.event_index DESC
+        ), solo_stackers AS (
+            SELECT DISTINCT ON (stacker) 
+                stacker, 
+                signer_key
+            FROM pox4_events
+            WHERE canonical = true AND microblock_canonical = true
+                AND name in ('stack-stx', 'stacks-increase', 'stack-extend')
+                AND start_cycle_id <= ${args.cycle_number} 
+                AND end_cycle_id >= ${args.cycle_number}
```

Comment:
This one I'll want to double check or create a test for. Might be `>` for end. (Non inclusive style)

Reply:
Yes, I'm pretty sure all events have the end as non-inclusive (the unlock height cycle), so the end condition should be `end_cycle_id > ${args.cycle_number}` for all cases. _(for agg `== +1` is fine, but both would work)_

Reply:
Thanks for the verification ❤️